crates/bevy_animation/src/curve_path.rs

```diff
@@ -0,0 +1,159 @@
+    /// are used, and therefore the accuracy.
+    pub fn from_cubic_curve(curve: CubicCurve<P>, subdivisions: usize) -> CurvePath<P> {
+        let arc_lengths: Vec<f32> = curve
+            .iter_positions(subdivisions)
+            .scan((0.0, curve.position(0.0)), |state, x| {
+                state.0 += x.distance(state.1);
+                state.1 = x;
+
+                Some(state.0)
+            })
+            .collect();
+
+        CurvePath { curve, arc_lengths }
+    }
+
+    /// Returns the [`CubicCurve`] this path is based on.
+    pub fn curve(&self) -> &CubicCurve<P> {
+        &self.curve
+    }
+
+    /// Total length of the curve.
+    pub fn length(&self) -> f32 {
+        self.arc_lengths[self.arc_lengths.len() - 1]
+    }
+
+    /// Returns a 't' value corresponding to the `t` value on the underlying curve that matches the
+    /// provided `distance`.
+    pub fn t(&self, distance: f32) -> f32 {
+        // Get index with greatest value that is less than or equal to target distance.
+        let closest = self.arc_lengths.partition_point(|&x| x <= distance) - 1;
+
+        // Check if index's distance perfectly matches target distance, otherwise lerp between the
+        // index and its next neighbor.
+        if self.arc_lengths[closest] == distance {
+            (closest + 1) as f32 / (self.arc_lengths.len() + 1) as f32
+        } else {
+            let length_before = self.arc_lengths[closest];
+            let length_after = self.arc_lengths[closest + 1];
+            let segment_length = length_after - length_before;
+
+            let segment_fraction = (distance - length_before) / segment_length;
+
+            (closest as f32 + segment_fraction) / (self.arc_lengths.len() - 1) as f32
+                * self.curve.segments().len() as f32
+        }
+    }
+
+    /// Compute the position of a point on the curve at `distance`.
+    pub fn position(&self, distance: f32) -> P {
+        self.curve.position(self.t(distance))
+    }
+
+    /// Compute the first derivative at `distance`. This is the instantaneous velocity of a point on the cubic curve at `distance`.
+    pub fn velocity(&self, distance: f32) -> P {
+        self.curve.velocity(self.t(distance))
+    }
+
+    /// Compute the second derivative at `distance`. This is the instantaneous acceleration of a point on the cubic curve at `distance`.
+    pub fn acceleration(&self, distance: f32) -> P {
+        self.curve.acceleration(self.t(distance))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::f32::consts::{PI, TAU};
+
+    use bevy_math::{
+        cubic_splines::{CubicBezier, CubicGenerator},
+        vec2, Vec2,
+    };
+
+    use crate::curve_path::CurvePath;
+
+    const ERROR: f32 = 0.001;
+    const SUBDIVISIONS: usize = 256;
+    // The 0.551915 is taken from https://www.mechanicalexpressions.com/explore/geometric-modeling/circle-spline-approximation.pdf
+    const CIRCLE_POINTS: [[Vec2; 4]; 4] = [
```

Comment:
A circle is a very regular shape which will hide any algorithm error because all cubic segments will be equal to each other, and in particular of constant length.